lib/options_validator.js

```diff
@@ -7,6 +7,43 @@ const VALIDATION_LEVEL_NONE = 'none';
 const VALIDATION_LEVEL_WARN = 'warn';
 const VALIDATION_LEVEL_ERROR = 'error';
 
+/**
+ * Checks that arguments provided to an operation match that operation's arity.
+ * An error will be thrown if the operation is called with an incorrect number of arguments.
+ *
+ * @method
+ * @param {object} args Arguments supplied to the operation that is calling assertArity.
+ * @param {number} requiredArity The correct arity of the operation.
+ */
+function assertArity(args, requiredArity) {
+  // acceptable values for arity are 0, 1, and 2
+  if (requiredArity < 0 || requiredArity > 2) {
```

Comment:
Since we plan on eventually hiding this function, I don't think we should check the required arity at execution time. This should be an error that occurs when `build.arity(3)` is called.

Reply:
I'm not sure I understand what you mean. This error will occur if `arity(3).options({}).build()` is called. We won't check the required arity at execution time once we change the operations to use the `OperationBuilder`.

Reply:
Ah, so we're changing it later. Got it.